astropy/visualization/astropy_support.py

```diff
@@ -0,0 +1,38 @@
+from astropy.visualization.time import time_support
+from astropy.visualization.units import quantity_support
+from contextlib import contextmanager, ExitStack
+
+
+@contextmanager
+def astropy_support(*, quantity_support_kwargs=None, time_support_kwargs=None):
+    """
+    Enable support for plotting `astropy.units.Quantity` and `astropy.time.Time` instances in
+    matplotlib.
+
+    It can be used as a decorator or with a ``with`` statement.
```

Comment:
nit, but I'd like the examples to be presented in the same order they are introduced in this sentence.
(out of scope here, but I'm also realizing that `quantity_support` and `time_support`'s respective docstrings currently do not demonstrate the decorator style, although they both support it)

Reply:
Yes, that makes sense. I've made the correction.
Maybe after this PR, another issue can be opened to fix `quantity_support` and `time_support`, and have all three in the same module.

Reply:
I thinkg that moving code around is rarely worth the cost of making `git blame` less useful. However, +1 on doing a follow up to complete docstrings !